controllers/datadogagent/utils.go

```diff
@@ -1936,6 +1936,21 @@ func mergeAnnotationsLabels(logger logr.Logger, previousVal map[string]string, n
 	return mergedMap
 }
 
+func removeAnnotationLabel(vals map[string]string, valToRemove string) map[string]string {
+	_, labelExists := vals[valToRemove]
+	if labelExists {
+		newVals := make(map[string]string, len(vals)-1)
+		for key, value := range vals {
```

Comment:
nit - naming is confusing, `valToRemove` is compared to `key` while there is another loop variable named `value`.